clusterloader2/pkg/measurement/common/simple/wait_for_pods.go

```diff
@@ -74,39 +77,67 @@ func (*waitForRunningPodsMeasurement) Execute(config *measurement.MeasurementCon
 	time.AfterFunc(timeout, func() {
 		close(stopCh)
 	})
-	return summaries, waitForPods(config.ClientSet, namespace, labelSelector, fieldSelector, desiredPodCount, stopCh, true)
+	return summaries, waitForPods(config.ClientSet, namespace, labelSelector, fieldSelector, desiredPodCount, stopCh, true, waitForRunningPodsMeasurementName)
 }
 
 // Dispose cleans up after the measurement.
 func (*waitForRunningPodsMeasurement) Dispose() {}
 
-func waitForPods(clientSet clientset.Interface, namespace, labelSelector, fieldSelector string, desiredPodCount int, stopCh <-chan struct{}, log bool) error {
+const (
+	uninitialized = ""
+	increasing    = "inc"
+	decreasing    = "dec"
+	equal         = "eq"
+)
+
+func waitForPods(clientSet clientset.Interface, namespace, labelSelector, fieldSelector string, desiredPodCount int, stopCh <-chan struct{}, log bool, callerName string) error {
 	// TODO(#269): Change to shared podStore.
 	ps, err := measurementutil.NewPodStore(clientSet, namespace, labelSelector, fieldSelector)
 	if err != nil {
 		return fmt.Errorf("pod store creation error: %v", err)
 	}
 	defer ps.Stop()
 
-	var runningPodsCount int
+	var podsStatus measurementutil.PodsStartupStatus
+	actionStatus := uninitialized
+	var oldPods []*corev1.Pod
 	for {
 		select {
 		case <-stopCh:
-			return fmt.Errorf("timeout while waiting for %d pods to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running", desiredPodCount, namespace, labelSelector, fieldSelector, runningPodsCount)
+			return fmt.Errorf("timeout while waiting for %d pods to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running", desiredPodCount, namespace, labelSelector, fieldSelector, podsStatus.Running)
 		case <-time.After(defaultWaitForPodsInterval):
 			pods := ps.List()
-			runningPodsCount = 0
-			for i := range pods {
-				if pods[i].Status.Phase == corev1.PodRunning {
-					runningPodsCount++
+			podsStatus = measurementutil.ComputePodsStartupStatus(pods, desiredPodCount)
+			if actionStatus != uninitialized {
+				diff := measurementutil.DiffPods(oldPods, pods)
+				deletedPods := diff.DeletedPods()
+				if actionStatus != decreasing && len(deletedPods) > 0 {
+					glog.Errorf("%s: %d pods disappeared for namespace(%s), labelSelector(%s), fieldSelector(%s): %v", callerName, len(deletedPods), namespace, labelSelector, fieldSelector, strings.Join(deletedPods, ", "))
+					glog.Infof("%s: %v", callerName, diff.String(sets.NewString()))
+				}
+				addedPods := diff.AddedPods()
+				if actionStatus != increasing && len(addedPods) > 0 {
+					glog.Errorf("%s: %d pods appeared for namespace(%s), labelSelector(%s), fieldSelector(%s): %v", callerName, len(deletedPods), namespace, labelSelector, fieldSelector, strings.Join(deletedPods, ", "))
+					glog.Infof("%s: %v", callerName, diff.String(sets.NewString()))
+				}
+			} else {
+				// Initialize action status.
```

Comment:
Remove this comment - it doesn't bring any value.

Reply:
Done.